### ISSUE TITLE
defaultDatabase number specified in redis connection string not honoured

### ISSUE THREAD

**Issue by chai-deshpande** [MichaCo/CacheManager#175]
I am trying to use the redis cache using the connection string approach and the defaultDatabase that I have specified in the connection string is not getting picked up when caching the data in redis.

My configuration looks like this:
```xml
  <cacheManager xmlns="http://cachemanager.michaco.net/schemas/CacheManagerCfg.xsd">
    <managers>
      <cache name="distributed-data-cache" enableStatistics="true" backplaneName="redis.cache" backplaneType="CacheManager.Redis.RedisCacheBackplane, CacheManager.StackExchange.Redis" serializerType="CacheManager.Serialization.Json.JsonCacheSerializer, CacheManager.Serialization.Json">
        <handle name="default" ref="inmemoryHandle" />
        <handle name="redis.cache" ref="redisHandle" isBackplaneSource="true" />
      </cache>
      <cache name="inmemory-data-cache" enableStatistics="true" serializerType="CacheManager.Serialization.Json.JsonCacheSerializer, CacheManager.Serialization.Json">
        <handle name="default" ref="inmemoryHandle" />
      </cache>
    </managers>
    <cacheHandles>
      <handleDef id="inmemoryHandle" type="CacheManager.SystemRuntimeCaching.MemoryCacheHandle`1, CacheManager.SystemRuntimeCaching" defaultExpirationMode="Sliding" defaultTimeout="2m" />
      <handleDef id="redisHandle" type="CacheManager.Redis.RedisCacheHandle`1, CacheManager.StackExchange.Redis" defaultExpirationMode="Sliding" defaultTimeout="5m" />
    </cacheHandles>
  </cacheManager>
```
Connection string is specified like this:
```xml
  <connectionStrings>
    <add name="redis.cache" connectionString="127.0.0.1:6379,allowAdmin=False,connectTimeout=500,defaultDatabase=3,ssl=False,abortConnect=False,connectRetry=3" />
  </connectionStrings>
```

With this configuration - I expect the data to be written to database 3 in redis. However, it is getting written to database 0.

**Comment by MichaCo** [comment]
Hi @chai-deshpande 
Thanks for reporting, yeah that's true, the `defaultDatabase` flag was recently added to the client library and  I never changed the configuration initialization to respect it. Most of the configuration options of CacheManager provide a way to define the database already, except using `<connectionStrings>` section...

Anyways, will be fixed in the next version. Until then, just use one of the other options, like CacheManager's redis configuration section...

Thanks,
Michael

**Comment by chai-deshpande** [comment]
Thanks @MichaCo 
Looking forward to the next release.

**Comment by mchudinov** [comment]
Thank you Michael! I'm waiting for this feature too.

**Comment by erizzo** [comment]
Is there an example of configuring this via code? I, too, need to specify a database other than db0.

**Comment by MichaCo** [comment]
fixed in release 1.1.2
